provider/oauth2.go

```diff
@@ -4,30 +4,39 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/MicahParks/keyfunc"
+	"github.com/pkg/errors"
 	"io"
 	"net/http"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/go-pkgz/rest"
 	"github.com/golang-jwt/jwt"
+	jwtv4 "github.com/golang-jwt/jwt/v4"
```

Comment:
is there a good reason to have both v4 and pre-gomod 3.x.x ? I think v4 was mostly compat with v3, at least i recall migrating some projects to v4 without any significant problem.

Reply:
That's what `keyfunc` is using, I don't think they have v3 compatible version anymore. If you are considering upgrading `auth` to `jwt/v4`, I'm happy to do that (as a separate PR) as well.